### PR TITLE
Extract utility and add unit tests for openapi tab indentation

### PR DESCRIPTION
**Pull request by develohpanda**
Follow-up to https://github.com/Kong/insomnia/pull/4315 to include a unit test. Extracts a utility and writes a unit test.

### CODE REVIEW

File: packages/insomnia-app/app/ui/components/codemirror/should-indent-with-tabs.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { CodeEditorProps } from './code-editor';
+
+export const shouldIndentWithTabs = ({ mode, indentWithTabs }: Pick<CodeEditorProps, 'mode' | 'indentWithTabs'>) => {
+  // YAML is not valid when indented with Tabs
+  const isYaml = mode?.includes('yaml') || false;
+
+  // OpenAPI is not valid when indented with Tabs
+  // TODO: OpenAPI in yaml is not valid with tabs, but in JSON is. Currently we do not differentiate and disable tabs regardless.
```
Comment (gatzjames) [inline packages/insomnia-app/app/ui/components/codemirror/should-indent-with-tabs.ts]: Is this something for this PR or should we make a ticket for it?

**Review by gatzjames — APPROVED** [review]
👍 Tested and looks good!
Not sure about the effort to allow json with tabs in openAPI so leaving it up to you!

File: packages/insomnia-app/app/ui/components/codemirror/should-indent-with-tabs.ts (hunk 1 above)
Comment (develohpanda) [reply packages/insomnia-app/app/ui/components/codemirror/should-indent-with-tabs.ts]: I'll probably make a ticket for it, not for this pr

Comment (develohpanda) [reply packages/insomnia-app/app/ui/components/codemirror/should-indent-with-tabs.ts]: INS-1390

Comment (develohpanda) [reply packages/insomnia-app/app/ui/components/codemirror/should-indent-with-tabs.ts]:
```suggestion
  // TODO: OpenAPI in yaml is not valid with tabs, but in JSON is. Currently we do not differentiate and disable tabs regardless. INS-1390
```

**Pull request merged**